pkg/addonmanager/controllers/addoninstall/controller_test.go

```diff
@@ -79,39 +89,89 @@ func TestReconcile(t *testing.T) {
 					t.Errorf("Should not install addon when controller is deleting")
 				}
 			},
-			testaddon: &testAgent{name: "test", strategy: agent.InstallAllStrategy("test")},
+			testaddons: map[string]agent.AgentAddon{
+				"test": &testAgent{name: "test", strategy: agent.InstallAllStrategy("test")},
+			},
 		},
 		{
 			name:                 "selector install strategy with unmatched cluster",
 			addon:                []runtime.Object{},
 			cluster:              []runtime.Object{addontesting.NewManagedCluster("cluster1")},
 			validateAddonActions: addontesting.AssertNoActions,
-			testaddon: &testAgent{name: "test", strategy: agent.InstallByLabelStrategy("test", metav1.LabelSelector{
-				MatchLabels: map[string]string{"mode": "dev"},
-			})},
+			testaddons: map[string]agent.AgentAddon{
+				"test": &testAgent{name: "test", strategy: agent.InstallByLabelStrategy("test", metav1.LabelSelector{
+					MatchLabels: map[string]string{"mode": "dev"},
+				})},
+			},
 		},
 		{
-			name:                 "selector install strategy with nil label selector",
```

Comment:
@qiujian16 The `labelSeletor` is removed from the InstallStrategy. So there is no need for this case.